### ISSUE TITLE
Picom crash on Awesome-Wm popup

### ISSUE THREAD

**Issue by Veids** [yshui/picom#535]
### Platform

Arch Linux amd64

### GPU, drivers, and screen setup

AMD Radeon 530, Mesa 20.2.1
```name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: X.Org (0x1002)
    Device: AMD STONEY (DRM 3.39.0, 5.9.3-arch1-1, LLVM 10.0.1) (0x98e4)
    Version: 20.2.1
    Accelerated: yes
    Video memory: 512MB
    Unified memory: no
    Preferred profile: core (0x1)
    Max core profile version: 4.5
    Max compat profile version: 4.5
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
Memory info (GL_ATI_meminfo):
    VBO free memory - total: 232 MB, largest block: 232 MB
    VBO free aux. memory - total: 2926 MB, largest block: 2926 MB
    Texture free memory - total: 232 MB, largest block: 232 MB
    Texture free aux. memory - total: 2926 MB, largest block: 2926 MB
    Renderbuffer free memory - total: 232 MB, largest block: 232 MB
    Renderbuffer free aux. memory - total: 2926 MB, largest block: 2926 MB
Memory info (GL_NVX_gpu_memory_info):
    Dedicated video memory: 512 MB
    Total available memory: 3584 MB
    Currently available dedicated video memory: 232 MB
OpenGL vendor string: X.Org
OpenGL renderer string: AMD STONEY (DRM 3.39.0, 5.9.3-arch1-1, LLVM 10.0.1)
OpenGL core profile version string: 4.5 (Core Profile) Mesa 20.2.1
OpenGL core profile shading language version string: 4.50
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.5 (Compatibility Profile) Mesa 20.2.1
OpenGL shading language version string: 4.50
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile

OpenGL ES profile version string: OpenGL ES 3.2 Mesa 20.2.1
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```

### Environment

Awesome-wm from git

### picom version

```
[ 11/04/2020 16:56:47.321 get_cfg WARN ] Dual-kawase blur is not implemented by the legacy backends, you must use the `experimental-backends` option.
[ 11/04/2020 16:56:47.482 init_render WARN ] Old backends only support blur method "kernel". Your blur setting will not be applied
**Version:** vgit-248bf

### Extensions:

* Shape: Yes
* XRandR: Yes
* Present: Present

### Misc:

* Use Overlay: Yes
* Config file used: /home/veids/.config/picom/picom.conf

### Drivers (inaccurate):

AMDGPU, Radeon
```

### Configuration:
```
# #   _________  __  ______  __________  __ __
# #	 / ___/ __ \/  |/  / _ \/_  __/ __ \/ |/ /
# #	/ /__/ /_/ / /|_/ / ___/ / / / /_/ /    /
# #	\___/\____/_/  /_/_/    /_/  \____/_/|_/
# #

# Rounded corners
corner-radius:          30;

# Shadows
shadow                  : false;
xinerama-shadow-crop    : true;
shadow-radius           : 0.1;
shadow-offset-x         : -5;
shadow-offset-y         : -5;
shadow-opacity          : 0.2;

shadow-exclude = [
    "class_g ?= 'slop'",
    "class_g ?= 'Polybar'",
    "class_g ?= 'Visualizer'",
    # "class_g ?= 'rofi'",
    # i3wm titlebar shadows (tabbed/stacked)
    # i3wm hidden window shadows (tabbed/stacked)
    # "_NET_WM_STATE@:32a *= '_NET_WM_STATE_HIDDEN'",
    # "_NET_WM_WINDOW_TYPE@:a *= 'SPLASH'",
    # "_NET_WM_WINDOW_TYPE@:a *= 'NOTIFICATION'",
    "window_type *= 'menu'",
    "window_type = 'utility'",
    "window_type = 'dropdown_menu'",
    "window_type = 'popup_menu'"
];

# Fading
fading                  : true;
fade-delta              : 20;
fade-in-step            : 0.08;
fade-out-step           : 0.08;

# Blur
blur-background: true
blur-kern = "3x3box"
blur-method: "dual_kawase"
blur-strength: 11
blur-background-frame: false
blur-background-fixed: false
blur-background-exclude: [
  "window_type = 'dock'",
  "window_type = 'desktop'",
  "window_type = 'droupdown'",
  "class_g = 'vlc'",
  "class_i *= 'tray'"
]

# Other
backend                 : "glx";
vsync                   : true;
use-damage              : "false";
glx-copy-from-front     : false;
glx-no-rebind-pixmap    : true;
glx-no-stencil          : true;
unredir-if-possible     : true;
xrender-sync            : false;
xrender-sync-fence      : false;
mark-wmwin-focused      : true;
mark-ovredir-focused    : false;
detect-rounded-corners  : true;
detect-client-opacity   : true;
refresh-rate            : 60;
inactive-opacity        : 0.60;
active-opacity          : 0.8;
opacity-rule            :
[
  "100:class_g = 'i3lock'",
  # "100:class_g = 'Anki'",
  "100:class_g = 'KeePassXC'",
  "100:class_g = 'Google-chrome'",
  "100:class_g = 'firefox'",
  "100:class_g = 'llpp'",
  "100:class_g = 'Alacritty'",
  "100:class_g = 'vlc'",
  "100:class_g = 'Wine'",
  "85:class_g = 'Rofi'"
];

wintypes:
{
  dropdown_menu = { shadow = true; full-shadow = true; };
  popup_menu = { shadow = true; full-shadow = true;}  ;
  menu = { shadow = true; full-shadow = true; };
  normal = { full-shadow = true; };
  dock = { shadow = false; };
};
```

### Steps of reproduction

1. Trigger some awesome WM popup
2. Crash!

It crashes from time to time and i don't know why

### Current Behavior
Crashes

### Stack trace

```
gef➤  bt
#0  0x00007fec38894615 in raise () from /usr/lib/libc.so.6
#1  0x00007fec3887d862 in abort () from /usr/lib/libc.so.6
#2  0x00007fec3887d747 in __assert_fail_base.cold () from /usr/lib/libc.so.6
#3  0x00007fec3888cbf6 in __assert_fail () from /usr/lib/libc.so.6
#4  0x0000556bf46f77d5 in win_on_win_size_change (ps=ps@entry=0x556bf4ebf320, w=w@entry=0x556bf4ec76e0) at ../src/win.c:1080
#5  0x0000556bf46fa6f7 in win_process_update_flags (ps=ps@entry=0x556bf4ebf320, w=w@entry=0x556bf4ec76e0) at ../src/win.c:423
#6  0x0000556bf46f338a in refresh_windows (ps=0x556bf4ebf320) at ../src/picom.c:1374
#7  handle_pending_updates (ps=0x556bf4ebf320, loop=0x7fec38e68720) at ../src/picom.c:1422
#8  draw_callback_impl (loop=0x7fec38e68720, ps=0x556bf4ebf320, revents=<optimized out>) at ../src/picom.c:1449
#9  0x0000556bf46f4019 in draw_callback (loop=0x7fec38e68720, w=0x556bf4ebf3e0, revents=<optimized out>) at ../src/picom.c:1539
#10 0x00007fec38e5c033 in ev_invoke_pending () from /usr/lib/libev.so.4
#11 0x00007fec38e5f902 in ev_run () from /usr/lib/libev.so.4
#12 0x0000556bf46ef6f4 in session_run (ps=0x556bf4ebf320) at ../src/picom.c:2411
#13 main (argc=<optimized out>, argv=<optimized out>) at ../src/picom.c:2513
```

**Comment by tryone144** [comment]
Looking at the stack trace, this is most likely a duplicate of #525.

**Comment by yshui** [comment]
I agree with @tryone144.

This should have been fixed by #546. If not, be sure to let us know here. @Veids
